webapp/src/main/java/io/github/microcks/web/RestController.java

```diff
@@ -225,9 +228,17 @@ public ResponseEntity<?> execute(
 
             // Adding other generic headers (caching directives and so on...)
             if (response.getHeaders() != null) {
+               EvaluableRequest evaluableRequest = MockControllerCommons.buildEvaluableRequest(body, resourcePath, request);
+               TemplateEngine templateEngine = TemplateEngineFactory.getTemplateEngine();
                for (Header header : response.getHeaders()) {
+                  // Render response header
+                  List<String> renderedHeader = new ArrayList<>();
+                  for (String val : header.getValues()) {
+                     renderedHeader.add(MockControllerCommons.renderResponseContent(evaluableRequest, templateEngine, val));
+                  }
+
```

Comment:
I think it could be nice to have this refactored into a dedicated `MockControllerCommons` utility method so that it could be more easily reused by other protocols as well. It's not on the path for the merge but much more a comment for future integration.